plugins/org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/validation/UIStrings.java

```diff
@@ -153,40 +146,16 @@ protected String referencesToString(Iterable<? extends JvmTypeReference> element
 	 * @since 2.4
 	 */
 	public String referenceToString(JvmTypeReference typeRef, String defaultLabel) {
-		if (typeRef != null) {
-			if (typeRef instanceof JvmAnyTypeReference)
-				return "Object";
-			else {
-				JvmType type = typeRef.getType();
-				if (type != null && type.eIsProxy() && typeRef.eResource() != null) {
-					URI proxyURI = EcoreUtil.getURI(type);
-					String fragment = proxyURI.fragment();
-					if (lazyURIEncoder.isCrossLinkFragment(typeRef.eResource(), fragment)) {
-						Triple<EObject, EReference, INode> decoded = lazyURIEncoder.decode(typeRef.eResource(), fragment);
-						INode node = decoded.getThird();
-						if (node != null) {
-							String text = node.getRootNode().getText();
-							ITextRegion textRegion = node.getTextRegion();
-							String result = text.substring(textRegion.getOffset(), textRegion.getLength() + textRegion.getOffset());
-							
-							if (result != null) {
-								int pointIndex = result.lastIndexOf('.');
-								if (pointIndex >= 0 && pointIndex < (result.length() - 1)) {
-									return result.substring(pointIndex + 1);
-								}
-							}
-							
-							return result;
-						} else {
-							return defaultLabel;
-						}
-					}
-				} else {
-					return typeRef.getSimpleName();
-				}
-			}
-		}
-		return defaultLabel;
+		if (typeRef == null)
```

Comment:
ok, I will add new tests. it was not just refactored, the method was bogus for handling unresolved type, e.g. for List<Unknown> it used to produce List<void> and so on.